pkg/ovsdb/condition.go

```diff
@@ -317,12 +317,77 @@ func (c *Condition) CompareEnum(row *map[string]interface{}) (bool, error) {
 	}
 }
 
+func extractInterfaceFromRow(row *map[string]interface{}, column, requestedInterfaceName string) (interface{}, error) {
+	var emptyInterface interface{}
+	switch colData := (*row)[column].(type) {
+	case string:
+		return colData, nil
+	case []interface{}:
+		if len(colData) != 2 {
+			return emptyInterface, fmt.Errorf("column data %v is not a (interfaceName,value) tuple (i.e. number of elements in column data != 2)", colData)
+		}
+		actualInterfaceName, ok := colData[0].(string)
+		if !ok {
+			return emptyInterface, fmt.Errorf("first element in column data %v is not a string (column data first element type: %T)", colData[0], colData[0])
+		}
+		if actualInterfaceName != requestedInterfaceName {
+			return emptyInterface, fmt.Errorf("column data worng interface name. requested interface name:%s, actual interface name :%s", requestedInterfaceName, actualInterfaceName)
+		}
+		actualInterface, ok := colData[1].([]interface{})
+		if !ok {
+			return emptyInterface, fmt.Errorf("second element in column data %v is not an []interface (column data second element type: %T)", colData[1], colData[1])
+		}
+		return actualInterface, nil
+	default:
+		return emptyInterface, fmt.Errorf("unsupported column data type %T (column data: %v)", colData, colData)
+	}
+}
+
+func extractSetFromRow(row *map[string]interface{}, column string) (libovsdb.OvsSet, error) {
+	setFromCol, err := extractInterfaceFromRow(row, column, "set")
+	if err != nil {
+		return libovsdb.OvsSet{}, err
```

Comment:
the same as for Map, you can return a pointer

Reply:
same as above:
this function was removed.
(we don't need it because we now using the `libovsdb.Row` unmarshaller on `prepareRow` )